app/api/routes.py

```diff
@@ -34,11 +34,21 @@ def resources():
 @requires_body
 @authenticate
 def post_resources():
-    validation_errors = validate_resource(request)
+    json = request.get_json()
```

Comment:
This feels awkward, having a single route leads to checking for types in multiple places. This makes it harder to encapsulate in tests as your cyclomatic complexity is higher.

Reply:
Yeah it felt awkward writing it... I'm just not sure how to break it out into two different routes when `POST /resources` already creates a single resource :-(

Reply:
I feel that something like `POST /resources-list` or `POST /batch-resources` wouldn't be very RESTful..

Reply:
https://stackoverflow.com/questions/411462/restful-way-to-create-multiple-items-in-one-request


Maybe changing `/resources` to `/resource` and have this be `resource-collection`

We could try something where we formalize what needs to be done to each resource, and then use an iterator on that resource that performs that action. 

But I'm not too sure I like how it makes the single item end up as. 
```python
def iter_resource_json(resources):
    if isinstance(resources, dict):
        yield resources
    elif isinstance(resources, list):
        for resource in resources:
            yield resource
            
            
def create_resources(json):
    
    
    errors = [validate_resource(resource) for resource in iter_resource_json(json['data'])]
    
    if any(error is not None for error in errors):
        return .....
        
    errors = [create_resource(resource) for resource in iter_resource_json(json['data'])]
```

Reply:
I'm actually considering @mega.mac.slice's suggestion of `POST /resources/bulk`. I like keeping it a separate endpoint because otherwise it feels like an awkward overload. Like when you're writing Java and trying to let a method take any type of input so you have like 50 different method signatures for all the permutations? Yeah, gross. I think I'm going to separate it into a different route, so it's really just down to what we name it. I like /resources/bulk if you like it

Reply:
Ended up making it only accept a list.